IonDotnet/Internals/Binary/RawBinaryWriter.cs

```diff
@@ -892,5 +897,30 @@ private void EnsureCapacity(int forIndex)
                 Array.Resize(ref _array, _array.Length * 2);
             }
         }
+
+        private ContainerType GetContainerType(IonType ionType)
+        {
+            switch (ionType)
+            {
+                case IonType.List:
+                    return ContainerType.List;
+                    break;
+                case IonType.Struct:
+                    return ContainerType.Struct;
+                    break;
+                case IonType.Sexp:
+                    return ContainerType.Sexp;
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        private bool IsNotContainerType(ContainerType currentContainerType)
```

Comment:
The name of this function sounds weird, can we name it as IsContainerType and call !IsContainerType on L336?